### PR TITLE
Refactor to bring XDLOps Tuning up to date

### PR DESCRIPTION
**Pull request by jerryyin**
Did minimal amount of refactoring possible to re-enable XDLops forward pass. This will setup a good baseline for the development of XDLOps tuning in all three directions.

### CODE REVIEW

**Comment by jerryyin** [comment]
Jenkins: retest this please

**Comment by jerryyin** [comment]
Jenkins: retest this please

File: mlir/include/mlir/Dialect/MIOpen/Tuning/GridwiseGemmParams.h [hunk 1]
```diff
@@ -834,4 +722,105 @@ class PopulateParams : public PopulateParamsBase {
                               int64_t &gemmCDstPerWrite, int64_t &gridSize);
 };
 
+class PopulateParamsXDL : public PopulateParamsBase {
+private:
+  llvm::SmallVector<InitParamsXDL, 4> initParameters = {
+      // M/block N/block K/block M/wave N/wave
+      {128, 128, 16, 64, 64},
```
Comment (whchung) [inline mlir/include/mlir/Dialect/MIOpen/Tuning/GridwiseGemmParams.h:729]: @jerryyin could you check with @zjing14 to ensure these initial vector looks correct?

File: mlir/include/mlir/Dialect/MIOpen/Tuning/GridwiseGemmParams.h [hunk 2]
```diff
@@ -834,4 +722,105 @@ class PopulateParams : public PopulateParamsBase {
...
+            threadGemmDataPerRead_GemmM, threadGemmDataPerRead_GemmN);
+
+    const auto a_block_space =
+        param->gemmKPerBlock *
+        integer_least_multiple(param->gemmMPerBlock, max_lds_align);
+    const auto b_block_space =
+        param->gemmKPerBlock *
+        integer_least_multiple(param->gemmNPerBlock, max_lds_align);
+
+    ldsSize = 2 * (a_block_space + b_block_space) * sizeof(float);
```
Comment (whchung) [inline mlir/include/mlir/Dialect/MIOpen/Tuning/GridwiseGemmParams.h:775]: @jerryyin for XDLOPS we don't do LDS double buffering.

**Review by whchung — APPROVED** [review]
LGTM. some old codes for older XDLOPS logic could be changed later on.

File: mlir/include/mlir/Dialect/MIOpen/Tuning/GridwiseGemmParams.h (hunk 1 above)
Comment (jerryyin) [reply mlir/include/mlir/Dialect/MIOpen/Tuning/GridwiseGemmParams.h:729]: Will add him in the next review.

File: mlir/include/mlir/Dialect/MIOpen/Tuning/GridwiseGemmParams.h (hunk 2 above)
Comment (jerryyin) [reply mlir/include/mlir/Dialect/MIOpen/Tuning/GridwiseGemmParams.h:775]: Will get rid of `2 *` in next review.

**Pull request merged**